core/src/mindustry/world/meta/StatValues.java

```diff
@@ -292,15 +292,15 @@ public static StatValue weapons(UnitType unit, Seq<Weapon> weapons){
                     continue;
                 }
 
-                TextureRegion region = !weapon.name.equals("") && weapon.region.found() ? weapon.region : Core.atlas.find("clear");
+                TextureRegion region = !weapon.name.isEmpty() ? Core.atlas.find(weapon.name + "-preview", weapon.region) : null;
```

Comment:
Is this fine, or should I put a `previewRegion` in `Weapon` (and change `Generators` accordingly to use `previewRegion` in weapon instead of `find`ing)?